### PR TITLE
[submit queue] Order by priority label then by PR number

### PR DESCRIPTION
**Pull request by eparis**

### CODE REVIEW

File: mungegithub/mungers/submit-queue.go [hunk 1]
```diff
@@ -520,15 +520,41 @@ func (sq *SubmitQueue) flushGithubE2EQueue(reason string) {
...
+
+	aPrio := a.Priority()
+	bPrio := b.Priority()
+	if aPrio < bPrio {
+		return true
+	} else if aPrio > bPrio {
+		return false
+	}
+
+	if *a.Issue.Number < *b.Issue.Number {
```
Comment (yujuhong) [inline mungegithub/mungers/submit-queue.go]: nit: return `*a.Issue.Number < *b.Issue.Number`

File: mungegithub/github/github.go [hunk 2]
```diff
@@ -539,6 +540,26 @@ func (obj *MungeObject) RemoveLabel(label string) error {
 	return nil
 }
 
+// Priority returns the priority an issue was labeled with.
+// The labels must take the form 'priority/P?[0-9]+'
+// or math.MaxInt32 if unset
+func (obj *MungeObject) Priority() int {
+	priority := math.MaxInt32
```
Comment (yujuhong) [inline mungegithub/github/github.go]: I was thinking whether a default priority such as p2 makes sense. This spares most people from applying priority labels explicitly.

Comment (pmorie) [inline mungegithub/github/github.go]: +1

File: mungegithub/mungers/submit-queue_test.go [hunk 3]
```diff
@@ -140,6 +141,82 @@ func FailJenkins() jenkins.Job {
...
+			name: "With a priority label",
+			issues: []github.Issue{
+				*github_test.Issue(whitelistUser, 2, []string{"priority/P1"}, true),
+				*github_test.Issue(whitelistUser, 3, []string{"priority/P1"}, true),
+				*github_test.Issue(whitelistUser, 4, []string{"priority/P0"}, true),
+				*github_test.Issue(whitelistUser, 5, nil, true),
+			},
+			expected: []int{4, 2, 3, 5},
+		},
+	}
```
Comment (pmorie) [inline mungegithub/mungers/submit-queue_test.go]: add some tests for trash labels

Comment (pmorie) [inline mungegithub/mungers/submit-queue_test.go]: add tests for p2/p3

File: mungegithub/github/github.go (hunk 2 above)
Comment (eparis) [inline mungegithub/github/github.go]: I really don't like that idea.  I don't want to punish someone for taking the time to approprietely prioritize something. I'd rather 'punish' the people who are lazy and provide no transparency...

**Comment by eparis** [comment]
Added more tests.  Decided to put 'unlabeled' as P3.  Not P2.

**Pull request merged**